plugins/rapid7_intsights/unit_test/test_get_iocs_by_filter.py

```diff
@@ -0,0 +1,86 @@
+                    "tags": ["MyTag_1"],
+                    "type": "Domains",
+                    "value": "rapid7.com",
+                    "whitelisted": "false",
+                    "reportedFeeds": [{"ID": "SampleID", "ConfidenceLevel": 3, "Name": "AlienVault OTX"}],
+                }
+            ],
+            "nextOffset": "2022-11-18T16:59:01.626Z",
+        }
+        cls.action = Util.default_connector(GetIocsByFilter())
+
+    @patch("requests.request", side_effect=Util.mock_request)
+    def test_get_indicator_by_filter_should_success(self, make_request):
+        actual = self.action.run({Input.LAST_UPDATED_FROM: "2000-12-31T00:00:00Z"})
+        self.assertEqual(self.expected, actual)
+
+    @patch("requests.request", side_effect=Util.mock_request)
+    def test_get_indicator_by_filter_should_success_all_inputs(self, make_request):
+        actual = self.action.run(
+            {
+                Input.LAST_UPDATED_FROM: "2000-12-31T00:00:00Z",
+                Input.LIMIT: 1000,
+                Input.TYPE: ["Domains"],
+                Input.LAST_UPDATED_TO: "2000-12-31T00:00:00Z",
+                Input.LAST_SEEN_FROM: "2000-12-31T00:00:00Z",
+                Input.LAST_SEEN_TO: "2000-12-31T00:00:00Z",
+                Input.FIRST_SEEN_FROM: "2000-12-31T00:00:00Z",
+                Input.FIRST_SEEN_TO: "2000-12-31T00:00:00Z",
+                Input.STATUS: "Active",
+                Input.SEVERITY: ["High", "Low"],
+                Input.SOURCE_IDS: ["123450000012345000001233"],
+                Input.KILL_CHAIN_PHASES: ["Exploitation", "Installation"],
+                Input.OFFSET: "2022-11-18T16:59:01.626Z",
+            }
+        )
+        self.assertEqual(self.expected, actual)
+
+    @patch("requests.request", side_effect=Util.mock_request)
+    def test_get_indicator_by_filter_should_success_when_empty(self, make_request):
+        actual = self.action.run({Input.LAST_UPDATED_FROM: "2000-12-30T00:00:00Z"})
+        expected = {"content": []}
+        self.assertEqual(expected, actual)
+
+    @patch("requests.request", side_effect=Util.mock_request)
+    def test_get_indicator_by_filter_should_success_when_limit_changed(self, make_request):
+        actual = self.action.run({Input.LAST_UPDATED_FROM: "2000-12-31T00:00:00Z", Input.LIMIT: 1})
+        self.assertEqual(self.expected, actual)
+
+    @patch("requests.request", side_effect=Util.mock_request)
+    def test_get_indicator_by_filter_should_success_when_limit_lower(self, make_request):
+        actual = self.action.run({Input.LAST_UPDATED_FROM: "2000-12-31T00:00:00Z", Input.LIMIT: 0})
+        self.assertEqual(self.expected, actual)
+
+    @patch("requests.request", side_effect=Util.mock_request)
+    def test_get_indicator_by_filter_should_success_when_limit_higher(self, make_request):
+        actual = self.action.run({Input.LAST_UPDATED_FROM: "2000-12-31T00:00:00Z", Input.LIMIT: 1001})
+        self.assertEqual(self.expected, actual)
```

Comment:
Please use `parameterized` here 😄

Reply:
Have updated now, let me know if this is what you had in mind.